transformer_engine/pytorch/attention.py

```diff
@@ -5310,8 +5308,12 @@ def __init__(
         self.cp_global_ranks = cp_global_ranks
         self.cp_stream = cp_stream
 
-        self.hidden_size_per_attention_head = k_channels
-        self.v_channels = k_channels if v_channels is None else v_channels
+        self.hidden_size_per_attention_head_k = (
```

Comment:
Is this unused? How could `self.hidden_size_per_attention_head` and `self.v_channels` change to `self.hidden_size_per_attention_head_k` and `self.hidden_size_per_attention_head_v` without any other changes in the code?

Reply:
It wasn't used, until I added that shape check later (see [here](https://github.com/cyanguwa/TransformerEngine/blob/ed8ed8ccbb941f94906106ee372566337d2d69d0/transformer_engine/pytorch/attention.py#L5635)). I can delete all this together but I think it's good to have that shape check as well.